src/openfermion/measurements/_hamiltonian_stabilizers_sets.py

```diff
@@ -0,0 +1,134 @@
+    The missing strings overlap with all stabilizers, and will need
+    to be measured even when an error occured.
+
+    Args:
+        Hamiltonian(QubitOperator): Hamiltonian to be split.
+        subsets(list): List of QubitOperators in which the Hamiltonian
+            has been split into.
+    Return:
+        left_paulis(QubitOperator): Only if there are left strings.
+            If no missing Paulis, returns an empyt QubitOperator.
+    """
+    # This functions uses python object sets to compare missing tuples.
+    # Hamiltonian keys as a set.
+    ham_set_form = set(hamiltonian.terms.keys())
+
+    # Hamiltonian subsets as a single set object.
+    aux_set = set()
+    for sb in subsets:
+        aux_set.update(set(sb.terms.keys()))
+
+    # Find any missing tuple in the Hamiltonian set compared to
+    # all subsets.
+    left_set = ham_set_form - aux_set
+
+    # Save missing Pauli strings in QubitOperator
+    # If no missing Paulis, returns 0
+    left_paulis = QubitOperator()
+    for p in left_set:
+        left_paulis += QubitOperator(p, hamiltonian.terms[p])
+    return left_paulis
+
+
+def get_hamiltonian_subsets(hamiltonian, stabilizers):
+    """
+    Create subsets of Hamiltonian.
+
+    This function splits a Hamiltonian into subsets of Pauli strings that do
+    not overlap with stabilizers.
+    The Pauli strings that overlap with all stabilizers will be stored with a
+    separate set.
+    It can be used to perform SV and not disregarding all data if only a single
+    stabilizers signals an error.
+
+    Args:
+        hamiltonian(QubitOperator): Hamiltonian to be split.
+        stabilizers(list or QubitOperator): List of stabilizers as
+            QubitOperators.
+    Return:
+        hamiltonian_subsets(list): List of QubitOperator which are a subset of
+            the Hamiltonian.
+            Each QubitOperator does not overlap with the stabilizer at the same
+            position in the list.
+        rest_paulis(QubitOperator): Operator that contains the Pauli
+            strings that overlap with all the stabilizers.
+            If there are no paulis left an empty QubitOperator is returned.
+    Raises:
+        TypeError: Input hamiltonian must be QubitOperator.
+        TypeError: Input stabilizer_list must be QubitOperator or list.
+    """
+    if not isinstance(hamiltonian, QubitOperator):
+        raise TypeError('Input terms must be QubitOperator.')
+    if not isinstance(stabilizers, (QubitOperator, list,
+                                    tuple, numpy.ndarray)):
+        raise TypeError('Input stabilizers must be QubitOperator or list.')
+
+    stabilizer_list = list(copy.deepcopy(stabilizers))
+    ham = copy.deepcopy(hamiltonian)
+
+    hamiltonian_subsets = []
+    for stab in stabilizer_list:
+        # Initialize subset
+        ham_subset = QubitOperator()
+        for pauli in ham:
+            if _check_stabilizer_overlap(pauli, stab) is False:
+                ham_subset += pauli
```

Comment:
The way you wrote this, a Pauli will be added to multiple subsets if it is compatible with them all. This seems surprising; if this is intentional then it should be clearly documented. For applications, do you need the Pauli to appear in every subset that it is compatible with, rather than just one of them?

Reply:
Thanks for the comment. This is a tricky bit because it is a matter of what the user expects from it.

My original idea was to split the Hamiltonian into subsets of Pauli operators that do not share qubits with a stabilizer (or symmetry) such that if an error is detected by measuring the stabilizer, some information can still be recovered. This allows one to extract information of a circuit run, although an error has occurred, as opposed to a naive error detecting scheme where the run will be disregarded all together.

If you have an opinion about it, please let me know. I will update the documentation accordingly.

Best,
Xavi